### PR TITLE
fix(server): manual workspace role assignment on project invite create + more bugfixes

### PR DESCRIPTION
**Pull request by fabis94**
* Instead of mapping project roles to workspace roles, we default to a Guest workspace role or allow specifying it through a new mutation
* Fixes in OIDC auth strategy & serverRole assignment in stream invites

### CODE REVIEW

File: packages/server/modules/auth/strategies/oidc.js [hunk 1]
```diff
@@ -97,21 +101,22 @@ module.exports = async (app, session, sessionStorage, finalizeAuth) => {
...
+          if (serverInfo.inviteOnly && !token) {
             throw new Error('This server is invite only. Please provide an invite id.')
           }
 
           // validate the invite
-          const validInvite = await validateServerInvite(
-            user.email,
-            req.session.inviteId
-          )
+          const validInvite = await validateServerInviteFactory({
```
Comment (fabis94) [inline packages/server/modules/auth/strategies/oidc.js:109]: This is the OIDC bug: The service invocation referred to the old service fn

File: packages/server/modules/auth/tests/integration/registration.spec.ts [hunk 2]
```diff
@@ -33,6 +37,22 @@ describe('Server registration', () => {
   let restApi: LocalAuthRestApiHelpers
   let apollo: TestApolloServer
 
+  const createInviteAsAdmin = async (
+    args: CreateProjectInviteMutationVariables | CreateServerInviteMutationVariables
+  ) => {
+    return await captureCreatedInvite(async () => {
```
Comment (alemagio) [inline packages/server/modules/auth/tests/integration/registration.spec.ts:43]: not a big deal, but as a general rule `return await` should be avoided

**Review by alemagio — APPROVED** [review]

File: packages/server/modules/auth/tests/integration/registration.spec.ts (hunk 2 above)
Comment (fabis94) [reply packages/server/modules/auth/tests/integration/registration.spec.ts:43]: Here's a reason for why I did it this way: https://jsfiddle.net/p2bz4j5x/

If you don't await the return, the parent function does not appear in the error stack trace

Comment (fabis94) [reply packages/server/modules/auth/tests/integration/registration.spec.ts:43]: image

**Pull request merged**